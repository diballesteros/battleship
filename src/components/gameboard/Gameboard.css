.board {
    display: flex;
    flex-flow: column;
    width: 550px;
    height: 550px;
    margin: auto;
}