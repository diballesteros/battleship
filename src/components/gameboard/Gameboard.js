import React, { Component } from 'react';
import './Gameboard.css';

import Square from './square/Square';

class Gameboard extends Component {
    constructor(props) {
        super(props);

        const initialBoard = new Array(100);
        for (var i = 0; i < 100; i++) {
          initialBoard[i] = {
            hit: false,
            position: i,
            type: 'ocean'
          };
        };

        this.state = {
            board : initialBoard
        }
    }

    updateBoard(i) {
      const modifiedBoard = this.state.board.slice();
      const testShip = {
        hit: true,
        position: i,
        type: 'ship'
      }
      modifiedBoard[i] = testShip;
      this.setState({
        board: modifiedBoard
      });
    }

    render() {
      return (
        <div className='board'>
          {this.state.board.map((square, i) =>
            <Square 
              key={i}
              type={this.state.board[i].type}
              hit={this.state.board[i].hit}
              hitHandler={() => this.updateBoard(i)}>
            </Square>)}
        </div>
      );
    }
}

export default Gameboard;
