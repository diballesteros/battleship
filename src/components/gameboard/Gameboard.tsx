import React from 'react';
import './Gameboard.css';
import Square from '../square/Square';
import * as Constants from '../../constants/constant';

const createInitialBoard = () => {
  const initialBoard = new Array(100);
  initialBoard.forEach((el, index) => {
    initialBoard[index] = {
      hit: false,
      position: index,
      type: 'ocean',
      id: null,
    };
  });
  return initialBoard;
};

const showShip = (type: any, isHit: boolean) => {
  if (type === 'ship' && isHit === true) {
    return 'ship';
  }
  return 'ocean';
};

const Gameboard: React.FC = ({
  ships,
  myBoard,
  receivePlayerAttack,
  resolveBoardDrop,
  playerMoves,
}: any) => {
  const board = createInitialBoard();

  const checkForHit = (i: number) => {
    return playerMoves.includes(i);
  };

  ships.forEach((ship: any) => {
    ship.positions.forEach((coordinate: any, index: number) => {
      board[coordinate] = {
        hit: ship.hits[index],
        position: coordinate,
        type: 'ship',
        id: ship.id,
      };
    });
  });

  return (
    <div>
      <div className="letter_row">
        {Constants.LETTERROW.map((value, i) => (
          <Square key={`letter-${i + 1}`} />
        ))}
      </div>
      <div className="lower_section">
        <div className="number_column">
          {Constants.NUMBERCOLUMN.map((value, i) => (
            <Square key={`number-${i + 1}`} />
          ))}
        </div>
        <div className="board">
          {myBoard
            ? board.map((square, i) => <Square key={`board-${i + 1}`} />)
            : board.map((square, i) => <Square key={`board-${i + 1}`} />)}
        </div>
      </div>
    </div>
  );
};

export default Gameboard;
