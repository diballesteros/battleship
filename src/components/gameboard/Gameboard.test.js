import React from 'react';
import { render } from '@testing-library/react';
import Gameboard from './Gameboard';

describe('Gameboard component', () => {
    test('Initialized board', () => {
        const testShips = [
            {
                id: 1,
                positions: [1, 2, 3]
            },
            {
                id: 2,
                positions: [9, 19, 29]
            }
        ];

        const { asFragment } = render(<Gameboard ships={testShips} />);

        expect(asFragment()).toMatchSnapshot();
    });
})
