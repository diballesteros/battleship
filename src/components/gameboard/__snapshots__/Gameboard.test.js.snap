// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gameboard component Initialized board 1`] = `
<DocumentFragment>
  <div
    class="board"
  >
    <div
      class="square ocean-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ship-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
    <div
      class="square ocean-square"
    />
  </div>
</DocumentFragment>
`;
